parser/parser.go

```diff
@@ -23,10 +25,17 @@ func (pr *ParseResult) GetFile(name string) *File {
 
 // ParseCodeRequest iterates through all the proto files in the code gen request, parses them, and finally adds them to
 // the returned ParseResult
-func ParseCodeRequest(req *plugin_go.CodeGeneratorRequest) *ParseResult {
+func ParseCodeRequest(req *plugin_go.CodeGeneratorRequest, excludePatterns []*regexp.Regexp) *ParseResult {
 	result := new(ParseResult)
 
+parseLoop:
 	for _, file := range req.GetProtoFile() {
+		for _, pattern := range excludePatterns {
+			// Skip all files that match pattern
+			if pattern.MatchString(*file.Name) {
```

Comment:
Rather than relying on the value being a `*string` a dereferencing it, I'd prefer to call `file.GetName()` here